### PR TITLE
feat(expo-cli): add package and bundle identifier to init

### LINKED ISSUE #432: Flags `--android-package` and `--ios-bundle-identifier` of `init` command not updating app.json
## Description

The `--android-package` and `--ios-bundle-identifier` flags of the `init` command do not set the corresponding properties in **app.json**.

## Expected Behavior

When passing the `--android-package com.expo.test` and `--ios-bundle-identifier com.expo.test` flags to the `init` command, the `expo.android.package` and `expo.ios.bundleIdentifier` properties, respectively, should both be set to `"com.expo.test"` all in **app.json**.

## Observed Behavior

When passing the `--android-package com.expo.test` and `--ios-bundle-identifier com.expo.test` flags to the `init` command, the `expo.android.package` and `expo.ios.bundleIdentifier` properties, respectively, do not appear at all in **app.json**.

## Environment

Output of `expo diagnostics`: 

>   Expo CLI 2.11.9 environment info:
>     System:
>       OS: Windows 10
>     Binaries:
>       Yarn: 1.13.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
>       npm: 6.5.0 - C:\Program Files\nodejs\npm.CMD
>     IDEs:
>       Android Studio: Version  3.3.0.0 AI-182.5107.16.33.5314842

### PR DESCRIPTION
**Pull request by byCedric**
Fixes #432

### CODE REVIEW

File: packages/expo-cli/src/commands/init.ts [hunk 1]
```diff
@@ -472,5 +523,7 @@ export default function(program: Command) {
     .option('--yarn', 'Use Yarn to install dependencies. (default when Yarn is installed)')
     .option('--name [name]', 'The name of your app visible on the home screen.')
     .option('--yes', 'Use default options. Same as "expo init . --template blank')
+    .option('--android-package [name]', 'The package name for your Android standalone app')
+    .option('--ios-bundle-identifier [name]', 'The bundle identifier for your iOS standalone app')
```
Comment (EvanBacon) [inline packages/expo-cli/src/commands/init.ts:527]: Maybe something like this 🤔 
```suggestion
    .option('--bundle-id [name]', 'The bundle identifier for your iOS standalone app')
```

File: packages/expo-cli/e2e/__tests__/init-test.ts [hunk 2]
```diff
@@ -23,8 +23,32 @@ test('init', async () => {
...
     { cwd, env: { ...process.env, YARN_CACHE_FOLDER: path.join(cwd, 'yarn-cache') } }
   );
-  expect(stdout).toMatch(`Your project is ready at ${cwd}`);
+  expect(stdout).toContain(`Your project is ready`);
   const appJson = await JsonFile.readAsync(path.join(cwd, 'hello-world/app.json'));
   expect(appJson).toHaveProperty(['expo', 'name'], 'Hello');
   expect(appJson).toHaveProperty(['expo', 'slug'], 'hello-world');
 });
+
+test('init with android package and bundle identifier', async () => {
```
Comment (EvanBacon) [inline packages/expo-cli/e2e/__tests__/init-test.ts:32]: awesome 😍

File: packages/expo-cli/src/commands/init.ts (hunk 1 above)
Comment (byCedric) [reply packages/expo-cli/src/commands/init.ts:527]: Yeah, it's a bit long for ios 😄 We could do `--package`/`--package-name` and `--bundle-id`?

**Comment by byCedric** [comment]
Oh wait, I'll patch the start e2e test. I know whats up with that 😄

**Comment by brentvatne** [comment]
we should ensure that when a bare project is initialized that it uses the bundle id and package flags as well

**Comment by byCedric** [comment]
Yeah, I noticed the `--name` flag is also not picked up in bare. I'll add some additional e2e tests for bare.

**Comment by byCedric** [comment]
I'm doing a follow up PR for the e2e tests, but I noticed more stuff is going on in bare.

- The iOS `project.pbxproj` still contains `productName = "Hello World";`
- Android namespaces are still just the name, but lowercased.

Is this what you meant that needs to be tested @brentvatne?

**Comment by brentvatne** [comment]
@byCedric - perhaps we can reuse the @expo/config code that handles setting bundleIdentifier and package? we use this only in eject currently and it should handle the cases you mentioned

File: packages/expo-cli/src/commands/init.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo-cli/src/commands/init.ts:523]: Didn't realize it was already a feature, we should probably keep it how it was https://github.com/expo/expo-cli/issues/824

**Comment by EvanBacon** [comment]
fix https://github.com/expo/expo-cli/issues/824

**Comment by fson** [comment]
@byCedric can we ship this?

**Comment by byCedric** [comment]
@fson I need to reimplement this using `@expo/config`, I'm afraid it's not yet mergeable. I'll get going on this on Monday or maybe this weekend, sorry!

**Comment by brentvatne** [comment]
@EvanBacon plans to use the bundle id and package prompts that he wrote for eject here too

**Pull request closed without merging**